packages/atoms/src/components/kiwi-modal/kiwi-modal.tsx

```diff
@@ -106,19 +122,36 @@ export class KiwiModal {
             <div class="modal-body">
               <slot name="modal-body"></slot>
             </div>
-            {this.withFooter && (
+            {some([
+              this.cancelText,
+              this.previousText,
+              this.nextText,
+              this.okText,
+            ]) && (
               <div class="modal-footer">
                 {this.cancelText && (
                   <button class="btn btn-link" onClick={this.handleClose}>
                     {this.cancelText}
                   </button>
                 )}
-                <button
-                  class="btn btn-primary"
-                  onClick={this.handleConfirmation}
-                >
-                  {this.okText}
-                </button>
+                {this.previousText && (
+                  <button class="btn btn-link" onClick={this.handlePrevious}>
+                    {this.previousText}
+                  </button>
+                )}
+                {this.nextText && (
+                  <button class="btn btn-primary" onClick={this.handleNext}>
+                    {this.nextText}
+                  </button>
+                )}
+                {this.okText && (
+                  <button
+                    class="btn btn-primary"
+                    onClick={this.handleConfirmation}
+                  >
+                    {this.okText}
+                  </button>
+                )}
```

Comment:
Als erster Aufschlag ist das ok, allerdings hätte ich ein paar Ideen für ein späteres Refactoring.

Die Komponente macht meiner Ansicht nach zu viel. Die vielen optionalen props sind ein guter Indikator dafür. Angenommen demnächst kommt noch eine Funktionalität (z.bsp. das direkte Navigieren zw. Steps, oder das Verhindern von eben jenem) würden noch mehr props hinzukommen und die UI-Logik würde noch komplexer. Das führt mit der Zeit zu _Eierlegenden-Wollmilch-Komponenten_ deren API unhandlich und unübersichtlich wird. Auch Änderungen an der Komponenten-Api sind so aufwändiger.

Eine Variante das zu lösen ist über Komposition. Vorstellbar wäre also ein generischeres Modal, welches dann von spezifischeren Modals zu neuer Funktionalität _komponiert_ wird. Dann hätte man sowas wie ein _ConfimationModal_ und ein _SteppedModal_ etc. Deren APIs wären dann schlanker und Kontextbezogener. Außerdem könnten so Änderungen an der API des generischen Modals an einer Stelle angepasst werden, statt in allen Konsumenten.